src/components/MasterDetails.js

```diff
@@ -138,25 +138,34 @@ export default class MasterDetails extends React.Component<MasterDetailsDefaultP
   state: MasterDetailsState;
 
   componentWillReceiveProps(newProps: MasterDetailsProps) {
-    const isRowPresent = newProps.rows.find((row) => {
-      return (row.id === this.state.detailsRowId);
-    });
-    if (!isRowPresent) {
+    if (newProps.rows.length === 0) {
+      this.setState({
+        selectedRows: [],
+        detailsRowId: '',
+      });
+    } else {
+      const ids = newProps.rows.map((row) => {
+        return row.id;
+      });
+      const newSelectedRows = this.state.selectedRows.filter((row) => { return ids.includes(row); });
       if (this.props.multiSelect) {
```

Comment:
Yes, what Ben said is correct... I think the `if` statement could be changed to `if (this.props.noEmptySelection)` instead because if it's set, you don't care about multi-select, you always want to make a selection.
Actually, you can change it to `if (this.props.noEmptySelection && newSelectedRows.length === 0)` and then you can push the first row's ID onto the newSelectedRows array. After that, outside of the `if` statement's block, you can determine the detailsRowId (it's the last item in the newSelectedRows array, unless the array is empty, in which case it's the empty string) and then update the state with these two values (at the same time, rather than calling `setState()` twice).